### PR TITLE
[tf.keras] Bugfix - Add Stateful Metrics to fit_generator

### PR DESCRIPTION
**Pull request by briannemsick**
Description:

This should address the 2nd (of 3 issues) in https://github.com/tensorflow/tensorflow/issues/20529.

This is just a sync with Keras master branch also: https://github.com/keras-team/keras/blob/master/keras/engine/training_generator.py

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @qlzh727: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by qlzh727** [comment]
Ping @fchoolet for review again.

Btw, the lint is failing for this PR. Please please accordingly.

https://source.cloud.google.com/results/invocations/9ec5867b-8d7f-4d5d-b5a1-13d9cce2271a/targets

**Comment by briannemsick** [comment]
Looks like this was refactored, going to close.
